kratos/linear_solvers/fallback_linear_solver.h

```diff
@@ -0,0 +1,897 @@
+     * @return True if additional physical data is needed, false otherwise.
+     */
+    bool AdditionalPhysicalDataIsNeeded() override
+    {
+        return GetCurrentSolver()->AdditionalPhysicalDataIsNeeded();
+    }
+
+    /**
+     * @brief Provides additional physical data required by the solver.
+     * @details Some solvers may require a minimum degree of knowledge of the structure of the matrix.
+     * For example, when solving a mixed u-p problem, it is important to identify the row associated with v and p.
+     * Another example is the automatic prescription of rotation null-space for smoothed-aggregation solvers,
+     * which require knowledge of the spatial position of the nodes associated with a given degree of freedom (DOF).
+     * This function provides the opportunity to provide such data if needed.
+     * @param rA The sparse matrix.
+     * @param rX The solution vector.
+     * @param rB The right-hand side vector.
+     * @param rDoFSet The set of degrees of freedom.
+     * @param rModelPart The model part.
+     */
+    void ProvideAdditionalData(
+        SparseMatrixType& rA,
+        VectorType& rX,
+        VectorType& rB,
+        typename ModelPart::DofsArrayType& rDoFSet,
+        ModelPart& rModelPart
+        ) override
+    {
+        GetCurrentSolver()->ProvideAdditionalData(rA, rX, rB, rDoFSet, rModelPart);
+    }
+
+    /**
+     * @brief Add a linear solver to the collection.
+     * @details This function adds a linear solver to the collection without extending parameters.
+     * @param pSolver Pointer to the linear solver to be added.
+     */
+    void AddSolver(LinearSolverPointer pSolver)
+    {
+        // Increase the solvers vector
+        mSolvers.push_back(pSolver);
+    }
+
+    /**
+     * @brief Add a linear solver to the collection with additional parameters.
+     * @details This function adds a linear solver to the collection and extends the parameters.
+     * @param pSolver Pointer to the linear solver to be added.
+     * @param ThisParameters Parameters associated with the linear solver.
+     * @note This function extends parameters, but the interface for this is yet to be decided.
+     */
+    void AddSolver(
+        LinearSolverPointer pSolver,
+        const Parameters ThisParameters
+        )
+    {
+        // Increase the solvers vector
+        AddSolver(pSolver);
+
+        // Extend the parameters
+        FillParametersFromSolver(pSolver);
+    }
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    /**
+     * @brief Get the Solvers list.
+     * @return const std::vector<LinearSolverPointer>& Reference to the vector of solvers.
+     */
+    const std::vector<LinearSolverPointer>& GetSolvers() const
+    {
+        return mSolvers;
+    }
+
+    /**
+     * @brief Set the Solvers list.
+     * @param rSolvers A vector of LinearSolverPointer to set.
+     */
+    void SetSolvers(const std::vector<LinearSolverPointer>& rSolvers)
+    {
+        mSolvers = rSolvers;
```

Comment:
this doesn't update the internal `Parameters`

Reply:
Done